xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -383,25 +384,21 @@ where
         E: From<diesel::result::Error> + From<StorageError>,
     {
         tracing::debug!("Transaction beginning");
-        {
-            let connection = self.conn_ref();
-            let mut connection = connection.inner_mut_ref();
-            <Db as XmtpDb>::TransactionManager::begin_transaction(&mut *connection)?;
-        }
 
         let conn = self.conn_ref();
+        let _guard = conn.start_transaction::<Db>()?;
```

Comment:
now that async transaction are gone and our connection is a mutex, why can't we use diesels [transaction](https://docs.rs/diesel/latest/diesel/connection/trait.Connection.html#method.transaction) function?